doc/modules/calibration.rst

```diff
@@ -73,10 +76,93 @@ to 0 or 1 typically.
 .. currentmodule:: sklearn.svm
 
 Linear Support Vector Classification (:class:`LinearSVC`) shows an even more
-sigmoid curve as the RandomForestClassifier, which is typical for
-maximum-margin methods (compare Niculescu-Mizil and Caruana [1]_), which
-focus on hard samples that are close to the decision boundary (the support
-vectors).
+sigmoid curve than :class:`RandomForestClassifier`. This characteristic is
+typical of maximum-margin methods (compare Niculescu-Mizil and Caruana [4]_)
+which focus on hard samples that are close to the decision boundary (the
+support vectors).
+
+.. currentmodule:: sklearn.calibration
+
+Two approaches for performing calibration of probabilistic predictions are
+provided: a parametric approach based on Platt's sigmoid model and a
+non-parametric approach based on :ref:`isotonic regression <isotonic>`.
+Probability calibration should be done on new data not used for model fitting.
+The :class:`CalibratedClassifierCV` class uses an internal cross-validation
+procedure: for each split, the model is fit on the train samples and a
+calibrator is estimated on the output of the model on the validation samples.
+The probabilities predicted for the folds are then averaged. Already fitted
+classifiers can be calibrated by :class:`CalibratedClassifierCV` via the
+parameter ``cv="prefit"``. In this case, the user has to manually ensure that
+the datasets used for model fitting and calibration are disjoint.
+
+The following figures demonstrate the benefit of probability calibration. The
+first figure presents a dataset with 2 classes and 3 blobs of data. The blob in
+the middle contains random samples of each class. The probability for the
+samples in this blob should be close to 0.5.
```

Comment:
Could you label the x and y axis in the graph?

Reply:
This paragraph is also really useful - maybe it could be added to `plot_calibration.py` ?

Reply:
Not sure I understand what you mean by labeling the axis of the graph showing the blobs ? They don't have any specific meaning.... but I maybe missing something !

Reply:
Maybe 'Feature 1' and 'Feature 2' ? Just to make it obvious/clear to new users?

In the example https://111067-843222-gh.circle-artifacts.com/0/doc/auto_examples/calibration/plot_calibration.html, I see that you explain creation of the dataset down in the code but it would be nice if you explained the graph of the blob dataset in the text. The text above would work.

Reply:
Ah sorry! I was just looking at the rendered docs and realise you didn't change much here. No worries, my problem is with this example not being clear enough. I guess that can be changed in separate PR